File: DS/stacks.c
//  Implementation of basic operations of a stack -- push,pop,peek

#include<stdio.h>
#include<stdlib.h>
int top = -1;
int stk[10];
void push(int num){
    if(top==10){
        printf("Stack full\n");
        return;
    }
    top++;
    stk[top]=num;
    printf("Element pushed\n");
}
void pop(){
    if(top==-1){
        printf("Stack empty\n");
        return;
    }
    printf("%d is being popped out",stk[top--]);
}
int peek(){
    if(top==-1){
        printf("Stack empty\n");
        return -1;
    }
    return stk[top];
}
int main(){
    int choice,num;
    while(1){
        printf("\n\nEnter your choice:\n1.Push\n2.Pop\n3.Peek\nAny other number to exit\n");
        scanf("%d",&choice);
        switch (choice)
        {
        case 1:printf("Enter the number you want to Push:");
               scanf("%d",&num);
               push(num); 
            break;
        case 2: pop();
            break;
        case 3: printf("Top element of the stack is: %d",peek()); 
            break;   
        default:
            exit(0);
        }
    }
    return 0;
}

File: DS/stacks/stacks-in-python/stacks.py
# Single array implementation of stacks
class Stack:
    def __init__(self,limit=10):
        self.stk =[]
        self.limit=limit
    def isEmptyStack(self):
        return len(self.stk)<=0 
    def push(self,data):
        if len(self.stk)>self.limit:
            print("Overflow")
        else:
            self.stk.append(data)
            print("Appended",self.stk)
    def pop(self):
        if len(self.stk) <= 0:
            print("Underflow! Cannot pop")
        else:
            print("Popped!")
            return self.stk.pop()
    def printStack(self):
        print("Stack:",self.stk) 
    def stackLength(self):
        return len(self.stk)           
    def peek(self):
        if len(self.stk) <= 0:
            print("Underflow!")
        else:
            return self.stk[-1]    
our_stack = Stack()
our_stack.push(5)   
our_stack.push(8)   
num = our_stack.pop()
print(num)
our_stack.printStack()
print(our_stack.pop())
our_stack.push(6)
our_stack.push(10)
our_stack.push(20)
print("Top element of the stack is",our_stack.peek())